src/stdlib/vma.bpf.c

```diff
@@ -0,0 +1,73 @@
+#define __KERNEL__
+#include <linux/types.h>
+
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_helpers.h>
+
+struct bpf_iter_task_vma {
+  __u64 __opaque[1];
+  // Make sure this structure is larger than the kernel.
+  __u64 reserved[4];
+};
+
+// In order to ensure that the custom BTF does not conflict with the kernel's
+// native BTF as much as possible, we try to avoid using vmlinux.h directly.
+// CO-RE is a good choice.
```

Comment:
Not sure I understand. Why can't we use `vmlinux.h`?

Reply:
I think we're just trying to avoid it when possible but I think it's preferable to just include it rather than defining these internal structs.

Reply:
How should we make the choice? I just changed CO-RE back to vmlinux.h. Looking forward to your reply.

Reply:
@Rtoax Sorry for the churn, I think we're still figuring this part out.

So if we include 'vmlinux.h' the codegen for this program below (taken from the runtime test) is 29372 lines long!
```
uprobe:tests/testprogs/uprobe_test:uprobeFunction1 { printf("0x%x -- 0x%x\n", *uaddr("GLOBAL_A"), *uaddr("GLOBAL_C")); exit(); }
```

Btw @Rtoax, both including vmlinux.h and your other version with the defined structs produce invalid BPF objects on llvm18 (and 17). I'm currently debugging a bit.